core/jvm/src/TimeZoneJvm.kt

```diff
@@ -51,24 +52,52 @@ public actual open class TimeZone internal constructor(internal val zoneId: Zone
     }
 }
 
-@Serializable(with = ZoneOffsetSerializer::class)
-public actual class ZoneOffset internal constructor(zoneOffset: jtZoneOffset): TimeZone(zoneOffset) {
-    internal val zoneOffset get() = zoneId as jtZoneOffset
+@Serializable(with = FixedOffsetTimeZoneSerializer::class)
+public actual class FixedOffsetTimeZone
+public actual constructor(public actual val utcOffset: UtcOffset): TimeZone(utcOffset.zoneOffset) {
+    @Deprecated("Use utcOffset.totalSeconds", ReplaceWith("utcOffset.totalSeconds"))
+    public actual val totalSeconds: Int get() = utcOffset.totalSeconds
+}
 
+@Serializable(with = UtcOffsetSerializer::class)
+public actual class UtcOffset(internal val zoneOffset: jtZoneOffset) {
     public actual val totalSeconds: Int get() = zoneOffset.totalSeconds
+
+    override fun hashCode(): Int = zoneOffset.hashCode()
+    override fun equals(other: Any?): Boolean = other is UtcOffset && this.zoneOffset == other.zoneOffset
+    override fun toString(): String = zoneOffset.toString()
+
+    public actual companion object {
+        public actual fun parse(offsetString: String): UtcOffset = try {
+            jtZoneOffset.of(offsetString).let(::UtcOffset)
+        } catch (e: DateTimeException) {
+            throw DateTimeFormatException(e)
+        }
+    }
 }
 
-public actual fun TimeZone.offsetAt(instant: Instant): ZoneOffset =
-        zoneId.rules.getOffset(instant.value).let(::ZoneOffset)
+
+public actual fun TimeZone.offsetAt(instant: Instant): UtcOffset =
+        zoneId.rules.getOffset(instant.value).let(::UtcOffset)
 
 public actual fun Instant.toLocalDateTime(timeZone: TimeZone): LocalDateTime = try {
     java.time.LocalDateTime.ofInstant(this.value, timeZone.zoneId).let(::LocalDateTime)
 } catch (e: DateTimeException) {
     throw DateTimeArithmeticException(e)
 }
 
+public actual fun Instant.toLocalDateTime(utcOffset: UtcOffset): LocalDateTime = try {
+    java.time.LocalDateTime.ofInstant(this.value, utcOffset.zoneOffset).let(::LocalDateTime)
+} catch (e: DateTimeException) {
+    throw DateTimeArithmeticException(e)
+}
+
+
 public actual fun LocalDateTime.toInstant(timeZone: TimeZone): Instant =
         this.value.atZone(timeZone.zoneId).toInstant().let(::Instant)
 
+public actual fun LocalDateTime.toInstant(utcOffset: UtcOffset): Instant =
+        this.value.atZone(utcOffset.zoneOffset).toInstant().let(::Instant)
```

Comment:
Can be simplified to `this.value.toInstant(utcOffset.zoneOffset).let(::Instant)`